File: app/src/main/java/com/example/project01_group05/api/MangaAttributes.java
package com.example.project01_group05.api;

import java.util.Map;

public class MangaAttributes {

    private Map<String, String> title;

    public Map<String, String> getTitle() {
        return title;
    }

    public String getDisplayTitle() {

        if (title == null || title.isEmpty()) {
            return "Unknown Title";
        }

        // Prefer English when available
        if (title.containsKey("en") && title.get("en") != null) {
            return title.get("en");
        }

        // Otherwise use whichever localized title MangaDex provides
        for (String value : title.values()) {
            if (value != null && !value.isEmpty()) {
                return value;
            }
        }

        return "Unknown Title";
    }
}

File: app/src/main/java/com/example/project01_group05/api/MangaAttributes.kt
package com.example.project01_group05.api

data class MangaAttributes(
    val title: Map<String, String>? = null,
    val tags: List<TagObject>? = null,
    val description: Map<String, String>? = null,
    val status: String? = null,
    val year: Int? = null,
    val originalLanguage: String? = null,
    val contentRating: String? = null
) {

    fun getDisplayTitle(): String {
        if (title.isNullOrEmpty()) {
            return "Unknown Title"
        }

        val englishTitle = title["en"]
        if (!englishTitle.isNullOrEmpty()) {
            return englishTitle
        }

        for (value in title.values) {
            if (value.isNotEmpty()) {
                return value
            }
        }

        return "Unknown Title"
    }

    fun getTagNames(): List<String> {
        return tags?.mapNotNull { tag ->
            tag.attributes?.name?.get("en") ?: tag.attributes?.name?.values?.firstOrNull()
        } ?: emptyList()
    }

    fun getDisplayDescription(): String {
        if (description.isNullOrEmpty()) {
            return "No description available."
        }

        val englishDescription = description["en"]

        if (!englishDescription.isNullOrEmpty()) {
            return englishDescription
        }

        for (value in description.values) {
            if (value.isNotEmpty()) {
                return value
            }
        }

        return "No description available."
    }
}

data class TagObject(
    val id: String,
    val type: String,
    val attributes: TagAttributes?
)

data class TagAttributes(
    val name: Map<String, String>?
)
